### PR TITLE
Update constraints documentation

### PR DESCRIPTION
**Pull request by kgaillot**

### CODE REVIEW

File: doc/Pacemaker_Explained/en-US/Ch-Constraints.txt [hunk 1]
```diff
@@ -43,11 +42,13 @@ the score (possibly as part of a custom resource agent).
...
-There are two alternative strategies for specifying which nodes a
-resources can run on.  One way is to say that by default they can run
-anywhere and then create location constraints for nodes that are not
-allowed.  The other option is to have nodes "opt-in" --  to start with
-nothing able to run anywhere and selectively enable allowed nodes.
+'Location constraints' tell the cluster which nodes a resource can run on.
+
+There are two alternative strategies. One way is to say that, by default,
+resources can run anywhere, and then the location constraints specify nodes
+that are not allowed (an 'opt-out' cluster). The other way is to start with
```
Comment (beekhof) [inline doc/Pacemaker_Explained/en-US/Ch-Constraints.txt]: Should the relative preference for various nodes be covered at this point?

File: doc/Pacemaker_Explained/en-US/Ch-Constraints.txt [hunk 2]
```diff
@@ -88,19 +89,21 @@ indexterm:[Constraints,Location,node]
...
-a subset of nodes the resource is physically capable of running on
-can significantly boost performance when a large set of nodes are present.
-When pacemaker_remote is in use to expand the node count into the hundreds of
-nodes range, this option should be considered.
+|Whether Pacemaker should perform resource discovery (that is, check whether
+ the resource is already running) for this resource on this node. This should
+ normally be left as the default, so that rogue instances of a service can be
+ stopped when they are running where they are not supposed to be. However,
+ there are two situations where disabling resource discovery is a good idea:
+ when a service is not installed on a node, discovery might return an error;
```
Comment (beekhof) [inline doc/Pacemaker_Explained/en-US/Ch-Constraints.txt]: s/discovery/discovery of badly written written agents/ ?

File: doc/Pacemaker_Explained/en-US/Ch-Constraints.txt [hunk 3]
```diff
@@ -88,19 +89,21 @@ indexterm:[Constraints,Location,node]
...
+ and when Pacemaker Remote is used to scale a cluster to hundreds of nodes,
+ limiting resource discovery to allowed nodes can significantly boost
+ performance.
```
Comment (beekhof) [inline doc/Pacemaker_Explained/en-US/Ch-Constraints.txt]: "< b>It is up to you as an administrator to ensure that the service can /never/ be active on nodes where resource-discovery is disabled.</ b>"

File: doc/Pacemaker_Explained/en-US/Ch-Constraints.txt [hunk 4]
```diff
@@ -108,9 +111,9 @@ nodes range, this option should be considered.
   required.
 * +exclusive:+ Perform resource discovery for the specified resource only on
   this node (and other nodes similarly marked as exclusive). Multiple location
-  constraints using 'exclusive' discovery for the same resource across
+  constraints using exclusive discovery for the same resource across
```
Comment (beekhof) [inline doc/Pacemaker_Explained/en-US/Ch-Constraints.txt]: i think the quotes are correct here and below

File: doc/Pacemaker_Explained/en-US/Ch-Constraints.txt [hunk 5]
```diff
@@ -208,21 +211,22 @@ indexterm:[Resource,Constraints,Ordering]
...
 ====
 Ordering constraints affect 'only' the ordering of resources;
-they do not require that the resources be placed on the
+they do 'not' require that the resources be placed on the
 same node. If you want resources to be started on the same node
-'and' in a specific order, you need an ordering constraint 'and'
-a location constraint (see <<s-resource-colocation>>).
+'and' in a specific order, you need both an ordering constraint 'and'
+a colocation constraint (see <<s-resource-colocation>>), or
+alternatively, a group (see <<group-resources>>).
```
Comment (beekhof) [inline doc/Pacemaker_Explained/en-US/Ch-Constraints.txt]: One day, in all our free time, we should create a "needs" (or some better name) constraint which implies ordering and colocation.

File: doc/Pacemaker_Explained/en-US/Ch-Constraints.txt (hunk 1 above)
Comment (kgaillot) [inline doc/Pacemaker_Explained/en-US/Ch-Constraints.txt]: How so? It already says above that the highest score wins, and the table immediately below describes how a score works for a location constraint. Maybe some examples in the detailed sections below?

File: doc/Pacemaker_Explained/en-US/Ch-Constraints.txt (hunk 2 above)
Comment (kgaillot) [inline doc/Pacemaker_Explained/en-US/Ch-Constraints.txt]: Or more commonly a system agent. Maybe "discovery might return an error (properly written OCF agents will not, so this is usually only seen with other agent types)"

File: doc/Pacemaker_Explained/en-US/Ch-Constraints.txt (hunk 3 above)
Comment (kgaillot) [inline doc/Pacemaker_Explained/en-US/Ch-Constraints.txt]: Good point, will expand

File: doc/Pacemaker_Explained/en-US/Ch-Constraints.txt (hunk 4 above)
Comment (kgaillot) [inline doc/Pacemaker_Explained/en-US/Ch-Constraints.txt]: '+' makes sense

File: doc/Pacemaker_Explained/en-US/Ch-Constraints.txt (hunk 5 above)
Comment (kgaillot) [inline doc/Pacemaker_Explained/en-US/Ch-Constraints.txt]: trello'd

**Comment by kgaillot** [comment]
Added commit with recommended changes, and moved Resource Sets to its own section

**Pull request merged**